src/github_permission_manager_demotion/handler.py

```diff
@@ -0,0 +1,127 @@
+import json
+import boto3
+import requests
+from datetime import datetime
+from common.shared_functions import GitHubAuth, DEFAULT_REGION
+import os
+
+def is_last_org_owner(all_owners, user):
+    if len(all_owners) == 1 and user in all_owners:
+        return True
+    return False
+
+class GitHubPermissionRemover:
+    def __init__(self, requests_module=requests):
+        self.dynamodb = None
+        self.ssm_client = None
+        self.app_id = None
+        self.private_key = None
+        self.installation_id = None
+        self.auth_headers = None
+        self.requests = requests_module
+
+    def initialise_aws_clients(self):
+        self.dynamodb = boto3.resource('dynamodb', region_name=DEFAULT_REGION)
+        self.ssm_client = boto3.client('ssm', region_name=DEFAULT_REGION)
+
+    def get_all_parameters(self):
+        workspace = os.getenv('WORKSPACE')
+        self.app_id = self.get_ssm_parameter(f"/github_permission_manager_webhook/${workspace}_app_id")
+        self.private_key = self.get_ssm_parameter(f"/github_permission_manager_webhook/${workspace}_private_key")
+        self.installation_id = self.get_ssm_parameter(f"/github_permission_manager_webhook/${workspace}_installation_id")
+        self.auth_headers = self.get_token_to_access_github()
+
+    def get_token_to_access_github(self):
+        github_auth = GitHubAuth(self.private_key, self.app_id, self.installation_id)
+        access_token = github_auth.get_access_token()
+        return {
+            "Authorization": f"Bearer {access_token}",
+            "Accept": "application/vnd.github.v3+json"
+        }
+
+    def get_ssm_parameter(self, parameter_name):
+        parameter = self.ssm_client.get_parameter(Name=parameter_name, WithDecryption=True)
+        return parameter['Parameter']['Value']
+
+    def close_issue(self, repository, issue_number):
+        response = self.requests.patch(
+            f"https://api.github.com/repos/{repository}/issues/{issue_number}",
+            headers=self.auth_headers,
+            json={"state": "closed"}
+        )
+        print(f"Issue closed: {response.status_code}")
+
+    def post_comment_on_issue(self, repository, issue_number, comment_body):
+        data = {
+            "body": comment_body
+        }
+        response = requests.post(
+            f"https://api.github.com/repos/{repository}/issues/{issue_number}/comments",
+            headers=self.auth_headers,
+            json=data
+        )
+        print(f"Comment posted: {response.status_code}")
+
+    def get_all_org_owners(self, organization):
+        response = requests.get(
+            f"https://api.github.com/orgs/{organization}/members?role=admin",
+            headers=self.auth_headers
+        )
+        print(f"Request to get org owners: {response.status_code}")
+        org_owners = []
+        for owner in response.json():
+            print(f"Owner: {owner['login']}")
+            org_owners.append(owner['login'])
```

Comment:
can we use a list comprehension here and then print the list afterwards?
I think it might be slightly neater but happy to leave as is

Reply:
I've converted to a list comprehension and removed the print statement as it was intended for debugging.